protobuf/lib/src/protobuf/pb_map.dart

```diff
@@ -39,6 +39,35 @@ class PbMap<K, V> extends MapBase<K, V> {
     _wrappedMap[key] = value;
   }
 
+  @override
+  bool operator ==(other) {
+    if (identical(other, this)) {
+      return true;
+    }
+    if (other is! PbMap) {
+      return false;
+    }
+    if (other.length != length) {
+      return false;
+    }
+    if (other.hashCode != hashCode) {
+      return false;
+    }
+    for (final key in keys) {
+      if (other[key] != this[key]) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  int get hashCode {
+    return hashObjects(_wrappedMap.keys
```

Comment:
Ideally we should be able to share code with _FieldSet.hashCode and PbList.hashCode